controller/volume_rebuilding_controller.go

```diff
@@ -183,6 +197,52 @@ func (vbc *VolumeRebuildingController) enqueueVolumeForReplica(obj interface{})
 	vbc.enqueueVolume(v)
 }
 
+func (vbc *VolumeRebuildingController) enqueueVolumeNodeReady(old, cur interface{}) {
+	oldNode, ok := old.(*longhorn.Node)
+	if !ok {
+		utilruntime.HandleError(fmt.Errorf("received unexpected old object: %#v", old))
+		return
+	}
+	curNode, ok := cur.(*longhorn.Node)
+	if !ok {
+		utilruntime.HandleError(fmt.Errorf("received unexpected current object: %#v", cur))
+		return
+	}
+
+	if curNode.Status.Conditions == nil {
+		return
+	}
+
+	oldReadyStatus := types.GetCondition(oldNode.Status.Conditions, longhorn.NodeConditionTypeReady).Status
+	curReadyStatus := types.GetCondition(curNode.Status.Conditions, longhorn.NodeConditionTypeReady).Status
+	oldSchedulableStatus := types.GetCondition(oldNode.Status.Conditions, longhorn.NodeConditionTypeSchedulable).Status
+	curSchedulableStatus := types.GetCondition(curNode.Status.Conditions, longhorn.NodeConditionTypeSchedulable).Status
+	if (oldNode.Spec.AllowScheduling == curNode.Spec.AllowScheduling || !curNode.Spec.AllowScheduling) &&
+		(oldReadyStatus == curReadyStatus || curReadyStatus == longhorn.ConditionStatusFalse) &&
+		(oldSchedulableStatus == curSchedulableStatus || curSchedulableStatus == longhorn.ConditionStatusFalse) {
+		return
```

Comment:
The condition is a bit dense. Consider extracting to make the intent clearer.
```suggestion

	readyChanged := oldReadyStatus == curReadyStatus || curReadyStatus == longhorn.ConditionStatusFalse
	schedulableChanged := oldSchedulableStatus == curSchedulableStatus || curSchedulableStatus == longhorn.ConditionStatusFalse
	schedulingChanged := oldNode.Spec.AllowScheduling == curNode.Spec.AllowScheduling || !curNode.Spec.AllowScheduling

	if !readyChanged && !schedulableChanged && !schedulingChanged {
		return
```

Reply:
Modified. Thanks.